### DISCUSSION TITLE
Tool change from tool table (no touch off)

### DISCUSSION THREAD

**Discussion by andrewmarles** [grblHAL/core#67] [Q&A]
Sorry, I asked this on the old repo by mistake.  I was searching through the old discussions.

With quick change tooling I would like to set up the tool table in GRBLHAL and then manually (or with ATC someday) change tools and have it apply the tool offset from the table. For example, set tool 1 as my probe/edge finder and use that to zero the work, and then have the tool change function apply offsets from there. However, looking at the semi-auto tool change modes, these are all wanting either a manual or automatic touch off and this is what the quick change tooling (TTS for example) allows you to skip since the tool offsets are pre-defined.

Anybody have the tool table working in GRBLHAL, or is this something we need to add?

**Comment by terjeio** [comment]
It is configurable - not extensively tested though...

https://github.com/grblHAL/core/blob/384fda6e3a135f6db1050f55a24e93260528111f/config.h#L480-L483

**Reply by andrewmarles** [reply]
Playing with this a bit today, it seems to work.  For future reference the sequence is as follows (this is the key info I was originally asking for):

1) Compile with the tool table enabled.  Define N_TOOLS.  Set $341 to 0 (Normal mode)
2) Issue the code T1 M6 to select tool 1 (usually probe or edge finder)
3) Install tool 1 and hit Cycle Start
4) Issue G43 to apply the offset from the tool table
5) Probe and zero WCS
6) Issue code T2 M6 to select tool 2
7) Install tool 2 and hit cycle start
8) Issue G43 to apply the offset from the tool table
9) Make chips

This is basically the normal sequence for tool changes, some slight changes would be needed in the post processor to move the spindle up to a convenient location for changing tools, but the basic functionality looks like it is there.
